// spoonacular_api/dist/com.spoonacular.client/com.spoonacular.client.model/InlineObject.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }(); /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * spoonacular API
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * The spoonacular Nutrition, Recipe, and Food API allows you to access over 380,000 recipes, thousands of ingredients, 80,000 food products, and 100,000 menu items. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * The version of the OpenAPI document: 1.0
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Contact: david@spoonacular.com
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      *
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      */

var _ApiClient = require('../ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
 * The InlineObject model module.
 * @module com.spoonacular.client/com.spoonacular.client.model/InlineObject
 * @version 1.0
 */
var InlineObject = function () {
  /**
   * Constructs a new <code>InlineObject</code>.
   * @alias module:com.spoonacular.client/com.spoonacular.client.model/InlineObject
   * @param ingredientList {String} The ingredient list of the recipe, one ingredient per line.
   * @param servings {Number} The number of servings.
   */
  function InlineObject(ingredientList, servings) {
    _classCallCheck(this, InlineObject);

    InlineObject.initialize(this, ingredientList, servings);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(InlineObject, null, [{
    key: 'initialize',
    value: function initialize(obj, ingredientList, servings) {
      obj['ingredientList'] = ingredientList;
      obj['servings'] = servings;
    }

    /**
     * Constructs a <code>InlineObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:com.spoonacular.client/com.spoonacular.client.model/InlineObject} obj Optional instance to populate.
     * @return {module:com.spoonacular.client/com.spoonacular.client.model/InlineObject} The populated <code>InlineObject</code> instance.
     */

  }, {
    key: 'constructFromObject',
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InlineObject();

        if (data.hasOwnProperty('ingredientList')) {
          obj['ingredientList'] = _ApiClient2.default.convertToType(data['ingredientList'], 'String');
        }
        if (data.hasOwnProperty('servings')) {
          obj['servings'] = _ApiClient2.default.convertToType(data['servings'], 'Number');
        }
        if (data.hasOwnProperty('defaultCss')) {
          obj['defaultCss'] = _ApiClient2.default.convertToType(data['defaultCss'], 'Boolean');
        }
        if (data.hasOwnProperty('showBacklink')) {
          obj['showBacklink'] = _ApiClient2.default.convertToType(data['showBacklink'], 'Boolean');
        }
      }
      return obj;
    }
  }]);

  return InlineObject;
}();

/**
 * The ingredient list of the recipe, one ingredient per line.
 * @member {String} ingredientList
 */


InlineObject.prototype['ingredientList'] = undefined;

/**
 * The number of servings.
 * @member {Number} servings
 */
InlineObject.prototype['servings'] = undefined;

/**
 * Whether the default CSS should be added to the response.
 * @member {Boolean} defaultCss
 */
InlineObject.prototype['defaultCss'] = undefined;

/**
 * Whether to show a backlink to spoonacular. If set false, this call counts against your quota.
 * @member {Boolean} showBacklink
 */
InlineObject.prototype['showBacklink'] = undefined;

exports.default = InlineObject;
