// spoonacular_api/dist/com.spoonacular.client/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DefaultApi = exports.InlineObject9 = exports.InlineObject8 = exports.InlineObject7 = exports.InlineObject6 = exports.InlineObject5 = exports.InlineObject4 = exports.InlineObject3 = exports.InlineObject2 = exports.InlineObject1 = exports.InlineObject = exports.ApiClient = undefined;

var _ApiClient = require('./ApiClient');

var _ApiClient2 = _interopRequireDefault(_ApiClient);

var _InlineObject = require('./com.spoonacular.client.model/InlineObject');

var _InlineObject2 = _interopRequireDefault(_InlineObject);

var _InlineObject3 = require('./com.spoonacular.client.model/InlineObject1');

var _InlineObject4 = _interopRequireDefault(_InlineObject3);

var _InlineObject5 = require('./com.spoonacular.client.model/InlineObject2');

var _InlineObject6 = _interopRequireDefault(_InlineObject5);

var _InlineObject7 = require('./com.spoonacular.client.model/InlineObject3');

var _InlineObject8 = _interopRequireDefault(_InlineObject7);

var _InlineObject9 = require('./com.spoonacular.client.model/InlineObject4');

var _InlineObject10 = _interopRequireDefault(_InlineObject9);

var _InlineObject11 = require('./com.spoonacular.client.model/InlineObject5');

var _InlineObject12 = _interopRequireDefault(_InlineObject11);

var _InlineObject13 = require('./com.spoonacular.client.model/InlineObject6');

var _InlineObject14 = _interopRequireDefault(_InlineObject13);

var _InlineObject15 = require('./com.spoonacular.client.model/InlineObject7');

var _InlineObject16 = _interopRequireDefault(_InlineObject15);

var _InlineObject17 = require('./com.spoonacular.client.model/InlineObject8');

var _InlineObject18 = _interopRequireDefault(_InlineObject17);

var _InlineObject19 = require('./com.spoonacular.client.model/InlineObject9');

var _InlineObject20 = _interopRequireDefault(_InlineObject19);

var _DefaultApi = require('./com.spoonacular/DefaultApi');

var _DefaultApi2 = _interopRequireDefault(_DefaultApi);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
* The_spoonacular_Nutrition_Recipe_and_Food_API_allows_you_to_access_over_380000_recipes_thousands_of_ingredients_80000_food_products_and_100000_menu_items__Our_food_ontology_and_semantic_recipe_search_engine_makes_it_possible_to_search_for_recipes_using_natural_language_queries_such_as_gluten_free_brownies_without_sugar_or_low_fat_vegan_cupcakes__You_can_automatically_calculate_the_nutritional_information_for_any_recipe_analyze_recipe_costs_visualize_ingredient_lists_find_recipes_for_whats_in_your_fridge_find_recipes_based_on_special_diets_nutritional_requirements_or_favorite_ingredients_classify_recipes_into_types_and_cuisines_convert_ingredient_amounts_or_even_compute_an_entire_meal_plan__With_our_powerful_API_you_can_create_many_kinds_of_food_and_especially_nutrition_apps_Special_diets_dietary_requirements_currently_available_include_vegan_vegetarian_pescetarian_gluten_free_grain_free_dairy_free_high_protein_whole_30_low_sodium_low_carb_Paleo_ketogenic_FODMAP_and_Primal_.<br>
* The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
* <p>
* An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
* <pre>
* var SpoonacularApi = require('com.spoonacular.client/index'); // See note below*.
* var xxxSvc = new SpoonacularApi.XxxApi(); // Allocate the API class we're going to use.
* var yyyModel = new SpoonacularApi.Yyy(); // Construct a model instance.
* yyyModel.someProperty = 'someValue';
* ...
* var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
* ...
* </pre>
* <em>*NOTE: For a top-level AMD script, use require(['com.spoonacular.client/index'], function(){...})
* and put the application logic within the callback function.</em>
* </p>
* <p>
* A non-AMD browser application (discouraged) might do something like this:
* <pre>
* var xxxSvc = new SpoonacularApi.XxxApi(); // Allocate the API class we're going to use.
* var yyy = new SpoonacularApi.Yyy(); // Construct a model instance.
* yyyModel.someProperty = 'someValue';
* ...
* var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
* ...
* </pre>
* </p>
* @module com.spoonacular.client/index
* @version 1.0
*/
/**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over 380,000 recipes, thousands of ingredients, 80,000 food products, and 100,000 menu items. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.0
 * Contact: david@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

exports.ApiClient = _ApiClient2.default;
exports.InlineObject = _InlineObject2.default;
exports.InlineObject1 = _InlineObject4.default;
exports.InlineObject2 = _InlineObject6.default;
exports.InlineObject3 = _InlineObject8.default;
exports.InlineObject4 = _InlineObject10.default;
exports.InlineObject5 = _InlineObject12.default;
exports.InlineObject6 = _InlineObject14.default;
exports.InlineObject7 = _InlineObject16.default;
exports.InlineObject8 = _InlineObject18.default;
exports.InlineObject9 = _InlineObject20.default;
exports.DefaultApi = _DefaultApi2.default;
